### PR TITLE
feat: add mechanism for serializing http error and refactor user module to use it

### PR DESCRIPTION
**Pull request by znikola**
closes: #3010

### CODE REVIEW

File: projects/core/src/occ/adapters/cart/occ-cart-entry.adapter.ts [hunk 1]
```diff
@@ -43,7 +42,7 @@ export class OccCartEntryAdapter implements CartEntryAdapter {
     return this.http
       .post<CartModification>(url, toAdd, { headers, params })
       .pipe(
-        catchError((error: any) => throwError(error.json())),
+        catchError((error: any) => throwError(error)),
```
Comment (dunqan) [inline projects/core/src/occ/adapters/cart/occ-cart-entry.adapter.ts]: Is this line changing anything?
It looks like we are we catching an error only to rethrow it again.

File: projects/core/src/occ/adapters/checkout/occ-checkout-delivery.adapter.ts [hunk 2]
```diff
@@ -99,7 +99,7 @@ export class OccCheckoutDeliveryAdapter implements CheckoutDeliveryAdapter {
         this.getCartEndpoint(userId) + cartId + '/deliverymodes'
       )
       .pipe(
-        catchError((error: any) => throwError(error.json())),
+        catchError(error => throwError(makeHttpErrorSerializable(error))),
```
Comment (dunqan) [inline projects/core/src/occ/adapters/checkout/occ-checkout-delivery.adapter.ts]: I would opt for not using `makeHttpErrorSerializable` in adapter.

File: projects/core/src/routing/routing.module.ts [hunk 3]
```diff
@@ -20,6 +21,7 @@ import { ROUTING_FEATURE } from './store/state';
     StoreModule.forFeature(ROUTING_FEATURE, reducerToken),
     EffectsModule.forFeature(effects),
     StoreRouterConnectingModule.forRoot({
+      routerState: RouterState.Minimal,
```
Comment (dunqan) [inline projects/core/src/routing/routing.module.ts]: Do we need this line if we use our custom serializer?

File: projects/core/src/user/store/effects/reset-password.effect.ts [hunk 4]
```diff
@@ -1,22 +1,17 @@
...
 
 @Injectable()
 export class ResetPasswordEffects {
   @Effect()
-  resetPassword$: Observable<
-    | fromActions.ResetPasswordSuccess
-    | AddMessage
-    | fromActions.ResetPasswordFail
-  > = this.actions$.pipe(
+  resetPassword$: Observable<Action> = this.actions$.pipe(
```
Comment (dunqan) [inline projects/core/src/user/store/effects/reset-password.effect.ts]: Any reason for making type weaker?

File: projects/core/src/occ/adapters/cart/occ-cart-entry.adapter.ts (hunk 1 above)
Comment (znikola) [reply projects/core/src/occ/adapters/cart/occ-cart-entry.adapter.ts]: Besides removing `.json()`, I didn't want to change anything as I'm not sure what's the idea behind rethrowing this error here.

File: projects/core/src/occ/adapters/checkout/occ-checkout-delivery.adapter.ts (hunk 2 above)
Comment (znikola) [reply projects/core/src/occ/adapters/checkout/occ-checkout-delivery.adapter.ts]: Good catch, reverted.

File: projects/core/src/routing/routing.module.ts (hunk 3 above)
Comment (znikola) [reply projects/core/src/routing/routing.module.ts]: Without it, this action fails:
Screen Shot 2019-06-19 at 09 06 17

File: projects/core/src/user/store/effects/reset-password.effect.ts (hunk 4 above)
Comment (znikola) [reply projects/core/src/user/store/effects/reset-password.effect.ts]: No reason, reverting

File: projects/core/src/occ/adapters/cart/occ-cart-entry.adapter.ts (hunk 1 above)
Comment (dunqan) [reply projects/core/src/occ/adapters/cart/occ-cart-entry.adapter.ts]: But by "only" removing ` .json()` you are changing everything that this line does ;-) So:
1) you should have the idea what this line does
2) why keep obsolete/redundant code that does nothing?

Comment (znikola) [reply projects/core/src/occ/adapters/cart/occ-cart-entry.adapter.ts]: We checked that `.json()` does not exist in the http error response, thus the removal and no need for me to dig deeper than that.

File: projects/core/src/routing/routing.module.ts (hunk 3 above)
Comment (dunqan) [reply projects/core/src/routing/routing.module.ts]: 👍

**Pull request merged**